charts/k8ssandra/templates/stargate/stargate.yaml

```diff
@@ -77,16 +77,16 @@ spec:
           resources:
             limits:
               cpu: {{ .Values.stargate.cpuLimMillicores | default 1000 }}m
-              memory: {{ mul 1.5 (.Values.stargate.heapMB | default 1024) }}Mi
+              memory: {{ mul 4 (.Values.stargate.heapMB | default 256) }}Mi
```

Comment:
Can we move the default into `values.yaml`?

This removes the need for the commented out documentation line (or even a placeholder which is the style used in the `prometheusSpec` block). My preference is:

```yaml
  prometheus:
    # Provisions an instance of Prometheus as part of this release
    enabled: true
    prometheusSpec:
      # Uncomment to listen for Prometheus traffic under a route prefix
      # externalUrl: http://localhost:9090/prometheus
      # routePrefix: /prometheus
```

but it may make sense to align behind the format used in https://github.com/norwoodj/helm-docs

Reply:
This dovetails into #258.

In general though I do agree that if we are setting defaults we should set them in value.yaml. In some cases it may not be possible though, like if the default is a calculated value.

Reply:
I have been following https://helm.sh/docs/chart_best_practices/values/ as a guide for how to document properties in `values.yaml`. 

What is not entirely clear to me is how best to document properties are commented out by default. Here is what we have in values.yaml for Stargate:

```yaml
stargate:
  # configuration regarding the Stargate API
  enabled: false
  replicas: 1
  clusterVersion: "3.11" # this is the only supported value for now
  containerImage: "auto" # chooses image based on clusterVersion
  # heapMB sets the heap size Stargate will use in megabytes. Default 256. Memory request and limit for the pod will be set to this value x2 and x4, respectively.
  # cpuReqMillicores sets the CPU request for the Stargate pod in millicores. default 200.
  # cpuLimMillicores sets the CPU limit for the Stargate pod in millicores. default 1000.
```
I think it could be easy to miss that those are properties. I think I would prefer to see this:

```yaml
stargate:
  # configuration regarding the Stargate API
  enabled: false
  replicas: 1
  clusterVersion: "3.11" # this is the only supported value for now
  containerImage: "auto" # chooses image based on clusterVersion
  # heapMB sets the heap size Stargate will use in megabytes. Default 256. Memory request and limit for the pod will be set to this value x2 and x4, respectively.
  # heapMB:
  
  # cpuReqMillicores sets the CPU request for the Stargate pod in millicores. default 200.
  # cpuReqMillicores:

  # cpuLimMillicores sets the CPU limit for the Stargate pod in millicores. default 1000.
  # cpuLimMillicores:
```
I realize that this is likely to highly subjective :)

Reply:
Makes sense to me, John. Chris's example includes reasonable values (possibly matching the default?) on the commented out lines. I think I like that better than leaving it blank. Thoughts?

Reply:
If we have default values that we are going to use, then yes definitely set them in `values.yaml`.